crypto/fipsmodule/evp/digestsign.c

```diff
@@ -218,7 +243,7 @@ int EVP_DigestSign(EVP_MD_CTX *ctx, uint8_t *out_sig, size_t *out_sig_len,
   FIPS_service_indicator_lock_state();
   int ret = 0;
 
-  if (uses_prehash(ctx, evp_sign)) {
+  if (uses_prehash(ctx, evp_sign) || used_for_hmac(ctx)) {
```

Comment:
Just an off-topic question. Doesn't this function update the service indicator counter even when the caller just wants the output length? Looks different than the behaviour in `EVP_DigestSignFinal()`, which this function also calls.

Reply:
Created `CryptoAlg-2226` for this